src/content-helper/dashboard-widget/components/top-posts.tsx

```diff
@@ -20,21 +22,95 @@ import {
 import { PostData } from '../../common/utils/post';
 import { DashboardWidgetProvider, TOP_POSTS_DEFAULT_LIMIT } from '../provider';
 import { TopPostListItem } from './top-posts-list-item';
-import { Telemetry } from '../../../js/telemetry/telemetry';
 
 const FETCH_RETRIES = 1;
 
+/**
+ * Defines the props structure for the TopPosts component.
+ *
+ * @since 3.13.0
+ */
+interface TopPostsProps {
+	settingsJson: string;
+}
+
+/**
+ * Defines the settings structure for the TopPosts component.
+ *
+ * @since 3.13.0
+ */
+interface TopPostsSettings {
+	period: Period;
+	metric: Metric;
+}
+
+/**
+ * Gets the settings from the passed JSON.
+ *
+ * If missing settings or invalid values are detected, they get set to their
+ * defaults.
+ *
+ * @since 3.13.0
+ *
+ * @param {string} settingsJson The JSON containing the settings.
+ *
+ * @return {TopPostsSettings} The resulting settings object.
+ */
+const getSettingsFromJson = ( settingsJson: string ): TopPostsSettings => {
+	const parsedSettings: TopPostsSettings = JSON.parse( settingsJson );
```

Comment:
I have a couple suggestions here:
 * Make `window.wpParselyContentHelperSettings` the default value of `settingsJson` (in the `TopPostsProps` component properties), so we don't need to explicitly set it with `<TopPosts settingsJson={ window.wpParselyContentHelperSettings } />` 
 *  in case `settingsJson` (aka `window.wpParselyContentHelperSettings`) is undefined for some reason, we could perhaps fallback to retrieving the user settings with a HTTP request to the API endpoint. This could add a little bit of complexity as it would require the function to be async, but might be worth to prevent any issues with the loading order of the scripts. 
 
If you think it makes sense, we can definitely tackle this down the line on a different PR. Let me know your thoughts.

Reply:
Thanks for the feedback!

- What I've done in the Sidebar (currently working on it) is accessing the settings directly in the component. So I think we could actually skip the whole prop situation here. Thoughts?
- My take is that I'd rather skip the complexity if I can. However, the function itself should definitely be resistant to `undefined` or malformed data. Technically, the endpoint returns defaults if something is broken, but this function is an extra protection if the endpoint isn't called or it happens to return junk data. So we should shield it as much as possible, and also see what happens if settings change form. So I think that there might be room for improvements here.

Whatever we do, let's do it in another PR. I'll leave this discussion open for further commenting.